### PR TITLE
Adding modifications for github actions to include all old Travis jobs.

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Added all old Travis jobs, except the `allow_failures` jobs.  Additionally, skipped an HTTP test that seems to be ill defined.  I do not think the big endian job is correctly designed.  I need to figure out how to use the correct image for this testing.

### CODE REVIEW

File: .github/workflows/asdf_ci.yml [hunk 1]
```diff
@@ -7,41 +7,141 @@ on:
...
+            toxenv: py39
+
+          - name: Mac OS Latest
+            os: macos-latest
+            python-version: 3.9
+            toxenv: py39
+
+          - name: Compatibility 
+            os: ubuntu-latest
+            python-version: 3.8
```
Comment (jdavies-st) [inline .github/workflows/asdf_ci.yml:61]: We can probably run the Compatibility, Bandit and Code Style and Twine checks on Python 3.9.

File: .github/workflows/asdf_ci.yml [hunk 2]
```diff
@@ -7,41 +7,141 @@ on:
...
+
+          - name: Bandit Security Checks
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: bandit
+
+          - name: Code Coverage
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: coverage
```
Comment (jdavies-st) [inline .github/workflows/asdf_ci.yml:72]: This is fine as a replacement for the current Travis CI matrix, but I think coverage should be folded into the `py39` test above.  And also perhaps the `py36` test above.  Running the exact same tests here just for coverage is redundant.

File: .github/workflows/asdf_ci.yml [hunk 3]
```diff
@@ -7,41 +7,141 @@ on:
...
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: twine
+
+          - name: Checkdocx
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: checkdocs
+
+          - name: Numpy 1.11
```
[the remark this answers is not in the document]
Comment (jdavies-st) [reply .github/workflows/asdf_ci.yml:89]: No.  Let's bump the minimum version.

File: .github/workflows/asdf_ci.yml [hunk 4]
```diff
@@ -7,41 +7,141 @@ on:
...
-      run: |
-        pytest
+      - name: Install System Packages
+        if: ${{ contains(matrix.toxenv,'docbuild') }}
+        run: |
+          sudo apt-get install graphviz texlive-latex-extra dvipng
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (jdavies-st) [reply .github/workflows/asdf_ci.yml:136]: Yeah, what's the effect of `fetch-depth`.  Does that mean it doesn't pull the tags?

File: .github/workflows/asdf_ci.yml [hunk 5]
```diff
@@ -7,41 +7,141 @@ on:
...
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: |
+          git submodule update --init --recursive
```
[the remark this answers is not in the document]
Comment (jdavies-st) [reply .github/workflows/asdf_ci.yml:143]: Also, shouldn't this submodule update be automatic?  If not, we should set it up so it is in the git repo config

[the remark this answers is not in the document]
Comment (jdavies-st) [reply .github/workflows/asdf_ci.yml:143]: Ah, you can also configure in the `checkout@v2` action itself:

```yaml
- uses: actions/checkout@v2
  with:
    submodules: true
```

File: .github/workflows/asdf_ci.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (jdavies-st) [reply .github/workflows/asdf_ci.yml:136]: Setting `fetch-depth: 0` causes the full history of the repo to fetched.  The default is a depth of 1, i.e. only the most recent commit.  The default means that dev tags won't be accurate, as one needs to pull in at a minimum the most recently-tagged commit.

So `fetch-depth: 0` might be overkill, but it is idiot-proof.

See https://github.com/actions/checkout#usage

[the remark this answers is not in the document]
Comment (eslavich) [reply .github/workflows/asdf_ci.yml:136]: Ah!  So this is all the things.  Okay, great.

File: .github/workflows/asdf_ci.yml [hunk 6]
```diff
@@ -7,41 +7,141 @@ on:
...
+          - name: Documentation Build
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: docbuild
+
+          - name: Big-endian platform
+            os: ubuntu-latest
+            python-version: 3.9
+            with: 
+              arch: s390x
```
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply .github/workflows/asdf_ci.yml:46]: I removed this from from the matrix and created a separate test for a big endian test.

File: .github/workflows/asdf_ci.yml [hunk 7]
```diff
@@ -7,41 +7,141 @@ on:
...
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: style
+
+          - name: Twine
+            os: ubuntu-latest
+            python-version: 3.8
+            toxenv: twine
+
+          - name: Checkdocx
```
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply .github/workflows/asdf_ci.yml:84]: I fixed this.

File: .github/workflows/asdf_ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply .github/workflows/asdf_ci.yml:89]: I removed the Numpy 1.11 test.

File: .github/workflows/asdf_ci.yml (hunk 1 above)
Comment (kmacdonald-stsci) [reply .github/workflows/asdf_ci.yml:61]: I changed these to 3.9.

File: .github/workflows/asdf_ci.yml (hunk 2 above)
Comment (kmacdonald-stsci) [reply .github/workflows/asdf_ci.yml:65]: I removed the py39 test and changed the code coverage test to 3.9.

File: .github/workflows/asdf_ci.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (kmacdonald-stsci) [reply .github/workflows/asdf_ci.yml]: I removed `git submodule update --init --recursive` and added the `with: submodules: true`.

**Pull request closed without merging**